text/0006-folio-distributed-vs-centralized-configration.md

```diff
@@ -0,0 +1,93 @@
+- Contributors:
+  - [Julian Ladisch](julian.ladisch@gbv.de)
+  - [Florian Gleixner](Florian.Gleixner@lrz.de)
+- RFC PRs: 
+  - PRELIMINARY REVIEW: https://github.com/folio-org/rfcs/pull/23
+  - DRAFT REVIEW: https://github.com/folio-org/rfcs/pull/28
+  - PUBLIC REVIEW: TBD
+  - FINAL REVIEW: TBD
+- Outcome: (Leave this blank.  Will eventually be either ACCEPTED or REJECTED)
+
+
+# Folio distributed vs. centralized configuration
+
+## Summary
+
+Drop mod-configuration module. Favour distributed configuration over centralized configuration, provide guidelines for storage of configuration values.
+
+## Motivation
+
+FOLIO relies on a tenant aware extendable microservice system. It is highly configurable and adoptable on all kind of requirements. FOLIO had no strict guidelines for developers, how and where to store the configuration values. This led to a situation where different approaches where developed by different developer teams.
+This makes it difficult for implementers, administrators and users to configure the system appropriate, as they may have to search the documentation how to read or write these configuration values.
+Basically there are two competing concepts where to store the values: distributed in the owning modules or centralized in a special module that offers a configuration store for other modules. There exist two solutions for centralized configurations: mod-settings and mod-configuration (deprecated, to be dropped).
+This RFC shall give guidelines for developers, where configuration values shall be set.
+
+## Detailed Explanation/Design
```

Comment:
There is not a lot of detail here. Will there be any kind of guidance about how individual modules should (must?) implement configuration, e.g. would it be possible to enforce (or at least recommend) a consistent pattern for API access? You have accurately described many of the problems with centralized storage in a microservices environment, but a major benefit of that pattern is the consistent API it provides. It would be extremely unfortunate if we were to move from the predictable interface of
```
/configurations/entries?query=module==${foo}
/configurations/entries?query=module==${bar}
```
to
```
/foo/conf
/bar/config
/bat/configuration
/baz/settings
```

Reply:
Distinguishing between configuration and data is not easy. Are circulation rules configuration or data?
We only can give a guide for the naming of API  endpoints for things that a module developer thinks that are configurations.
There are very few guidelines for the API. All I found is here: https://dev.folio.org/guidelines/naming-conventions/#api-endpoints
If we add guidelines in this RFC, it must be clear, that these guidelines shall become part of the mentioned page and that these guidelines should not generate changes to existing API endpoints.